youtube_dl/extractor/rte.py

```diff
@@ -60,3 +62,56 @@ def _real_extract(self, url):
             'thumbnail': thumbnail,
             'duration': duration,
         }
+
+
+
+class RteRadioIE(InfoExtractor):
+    IE_NAME = 'rte.ie'
+    IE_DESC = 'Raidió Teilifís Éireann Radio'
+    _VALID_URL = (
+        r'https?://(?:www\.)?rte\.ie/radio/utils/radioplayer/rteradioweb\.html'
+        r'#!rii=(?P<channel_id>[0-9]+)(?:%3A|:)'
+        r'(?P<id>[0-9]+)(?:%3A|:)'
+        r'(?P<pia>[0-9]+)(?:%3A|:)'
+        r'(?P<date>[0-9]{2}(%2D|-)[0-9]{2}(%2D|-)[0-9]{4})(?:%3A|:)'
+    )
+    _TEST = {
+        'url': 'http://www.rte.ie/radio/utils/radioplayer/rteradioweb.html#!rii=16:10507902:2414:27-12-2015:',
+        'info_dict': {
+            'id': '10507902',
+            'ext': 'flv',
+            'title': 'Gloria',
+            'thumbnail': 're:^https?://.*\.jpg$',
+            'description': 'Tim Thurston guides you through a millennium of sacred music featuring Gregorian chant, pure solo voices and choral masterpieces, framed around the glorious music of J.S. Bach.',
+            'duration': 7230.0,
+        },
+        'params': {
+            'skip_download': 'f4m fails with --test atm'
+        }
+    }
+
+    def _real_extract(self, url):
+        item_id = self._match_id(url)
+        feeds_url = 'http://www.rte.ie/rteavgen/getplaylist/?type=web&format=json&id=' + item_id
+        json_string = self._download_json(feeds_url, item_id)
+
+        show = json_string['shows'][0]
+        title, description, thumbnail = show['title'], show['description'], show['thumbnail']
+        duration = float_or_none(show['duration'], 1000)
+
+        f4m_url = show['media:group'][0]['hds_server'] + show['media:group'][0]['hds_url']
+        f4m_formats = self._extract_f4m_formats(f4m_url, item_id)
+        f4m_formats = [{
+            'format_id': f['format_id'],
+            'url': f['url'],
+            'ext': 'flv',
+        } for f in f4m_formats]
```

Comment:
It doesn't appear to work that way. When I leave out the 'ext' item the extension of the file written to disk is .f4m.